### PR TITLE
list only latest object if multiple versions exist

### PR DESCRIPTION
**Pull request by rhsakarpos**
**What this PR does / why we need it**:
If versioned objects exist, list only the latest versions in the listObject API

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
multiple object versions in DB
image

after listing only the latest versions
image

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by rhsakarpos** [comment]
@sfzeng @kumarashit @skdwriting @nguptaopensds @anvithks please review

File: s3/pkg/meta/db/drivers/tidb/bucket.go [hunk 1]
```diff
@@ -197,12 +201,13 @@ func (t *TidbClient) GetBuckets(ctx context.Context) (buckets []*Bucket, err err
 		if sseErr != nil {
 			return
 		}
-		sseType := "NONE"
-		if sseOpts != nil{
-			sseType = sseOpts.SseType
-		}
 		tmp.ServerSideEncryption = &pb.ServerSideEncryption{}
-		tmp.ServerSideEncryption.SseType = sseType
+		if sseOpts!=nil{
```
Comment (sfzeng) [inline s3/pkg/meta/db/drivers/tidb/bucket.go]: Hi @rhsakarpos Format the code please.

**Comment by sfzeng** [comment]
Hi @rhsakarpos ,  this seems what I want. But another function which name is CountObjects also needs to be changed, otherwise, the count and detail list will not match. The count is used to calculate the progress of the migration. Thanks.

File: s3/pkg/meta/db/drivers/tidb/bucket.go (hunk 1 above)
Comment (rhsakarpos) [reply s3/pkg/meta/db/drivers/tidb/bucket.go]: oops, done

**Comment by rhsakarpos** [comment]
> Hi @rhsakarpos , this seems what I want. But another function which name is CountObjects also needs to be changed, otherwise, the count and detail list will not match. The count is used to calculate the progress of the migration. Thanks.

this is fixed now, have pushed the changes

**Comment by sfzeng** [comment]
@rhsakarpos Please fix the conflict, thanks.

**Comment by rhsakarpos** [comment]
closing, since could not merge  > 10 days

**Pull request closed without merging**